substratevm/mx.substratevm/testhello.py

```diff
@@ -149,36 +168,75 @@ def test():
     checker.check(exec_string, skip_fails=False)
 
     # run a backtrace
-    # expect "#0  hello.Hello.main(java.lang.String[]).* at hello.Hello.java:67"
+    # expect "#0  hello.Hello.main(java.lang.String[] *).* at hello.Hello.java:67"
     # expect "#1  0x[0-9a-f]+ in com.oracle.svm.core.code.IsolateEnterStub.JavaMainWrapper_run_.* at [a-z/]+/JavaMainWrapper.java:[0-9]+"
     exec_string = execute("backtrace")
-    checker = Checker("backtrace hello.Hello.main",
-                      [r"#0%shello\.Hello\.main\(java\.lang\.String\[\]\)%s at hello/Hello\.java:67"%(spaces_pattern, wildcard_pattern),
-                       r"#1%s%s in com\.oracle\.svm\.core\.code\.IsolateEnterStub\.JavaMainWrapper_run_%s at %sJavaMainWrapper\.java:[0-9]+"%(spaces_pattern, address_pattern, wildcard_pattern, package_pattern)
+    checker = Checker("backtrace hello.Hello::main",
+                      [r"#0%shello\.Hello::main\(java\.lang\.String\[\] \*\)%s at hello/Hello\.java:67"%(spaces_pattern, wildcard_pattern),
+                       r"#1%s%s in com\.oracle\.svm\.core\.code\.IsolateEnterStub::JavaMainWrapper_run_%s at %sJavaMainWrapper\.java:[0-9]+"%(spaces_pattern, address_pattern, wildcard_pattern, package_pattern)
                        ])
     checker.check(exec_string, skip_fails=False)
 
+    if can_print_data:
+        # print the contents of the arguments array which will be in rdi
+        exec_string = execute("print /x *(('java.lang.String[]' *)$rdi)")
```

Comment:
When I run `mx debuginfotest` locally on 637d66a875578c991bb171fe720f41b32601d289 with gdb
```
GNU gdb (GDB) Fedora 10.1-2.fc33
```
gdb aborts when executing line 182:
```
gdb) list
67	        Greeter greeter = Greeter.greeter(args);

(gdb) backtrace
#0  hello.Hello::main(java.lang.String[] *) () at hello/Hello.java:67
#1  0x000000000041cef5 in com.oracle.svm.core.code.IsolateEnterStub::JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b(int, org.graalvm.nativeimage.c.type.CCharPointerPointer *) () at com/oracle/svm/core/JavaMainWrapper.java:146

(gdb) print /x *(('java.lang.String[]' *)$rdi)
gdb.error: Location address is not set.
```
Do I need another version of gdb (I though I read that 10.1 should be good)?

Reply:
Hmm, yes  10.1 is supposed to avoid this problem. I'll take a look at this once I fix the script as you recommended earlier.

Reply:
Well, that's disappointing. This appears to be a problem with the fedora build of gdb 10.1. When I run with a gdb 10.1 built from the vanilla gnu sources this error does not happen.

I think I know where this is going wrong (there is a patch in fedora's c-valprint.c that looks decidedly dodgy) but I need to check using a debuggable gdb build based on the fedora code. I'll post an update as soon as I can find what is wrong.

Reply:
I confirmed the problem is indeed a patch applied to valops.c which I have reported to Red Hat's gdb team. I'm not really clear what the patch is meant to achieve but it actually looks incoherent. The inserted code tests a pointer value to see if there is a data_location on target type for the pointer and special cases that situation to fetch the address by calling a routine that looks for a data_location on the type of the pointer value (i.e. the pointer type, not the target). That doesn't look at all right.

I'll let you know what the gdb team come up with. n.b. I have asked them if they can ensure any fix to this gets pushed back to RHEL and fedora 32 if that is needed.

Reply:
Awesome! Thanks for getting this fixed.